### PR TITLE
[4.0] RavenDB-11379 Fixed handling of attachment deletions in Raven ETL by…

### PR DESCRIPTION
**Pull request by arekpalinski**
… tracking attachment tombstones and sending AttachmentDELETE command

### CODE REVIEW

File: src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtl.cs [hunk 1]
```diff
@@ -41,6 +41,14 @@ protected override IEnumerator<RavenEtlItem> ConvertTombstonesEnumerator(IEnumer
             return new TombstonesToRavenEtlItems(tombstones, collection);
         }
 
+        protected override bool ShouldTrackAttachmentTombstones()
+        {
+            // if script isn't empty we relay on addAttachment() calls and detect that attachments needs be deleted
```
Comment (ayende) [inline src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtl.cs:46]: `relay` -> `rely`

File: src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtl.cs [hunk 2]
```diff
@@ -41,6 +41,14 @@ protected override IEnumerator<RavenEtlItem> ConvertTombstonesEnumerator(IEnumer
...
+            // when this call gets an attachment reference marked as null ($attachment/{attachment-name}/$null)
```
Comment (ayende) [inline src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtl.cs:47]: What happens if they do something like:

```
for(var attachment in this["@metadata"]["@attachments"])
{
   if(attachment.Name.endsWith(".png"))
       addAttachment(...);
}
```

In this manner, you don't _see_ that there are attachments to be deleted on the other side.

Comment (arekpalinski) [reply src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtl.cs:47]: Correct. Also it wound't work for renaming attachments. It means we always have to track them.

Comment (ayende) [reply src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtl.cs:47]: So, we need to delete all the other attachments that exists on destination when we send it, no?

Comment (arekpalinski) [reply src/Raven.Server/Documents/ETL/Providers/Raven/RavenEtl.cs:47]: True, that would make it easier.

**Pull request merged**